apps/rush-lib/src/api/CommandLineConfiguration.ts

```diff
@@ -29,270 +96,399 @@ export class CommandLineConfiguration {
-      ' graph for locally linked projects.  The number of simultaneous processes will be' +
-      ' based on the number of machine cores unless overridden by the --parallelism flag.' +
-      ' (For an incremental build, see "rush build" instead of "rush rebuild".)',
-    enableParallelism: true,
-    ignoreMissingScript: false,
-    ignoreDependencyOrder: false,
-    incremental: false,
-    allowWarningsInSuccessfulBuild: false,
-    safeForSimultaneousRushProcesses: false
-  };
+  private _additionalPathFolders: string[] = [];
 
   /**
    * Use CommandLineConfiguration.loadFromFile()
    *
    * @internal
    */
   public constructor(commandLineJson: ICommandLineJson | undefined) {
-    if (commandLineJson) {
-      if (commandLineJson.phases) {
-        for (const phase of commandLineJson.phases) {
-          if (this.phases.has(phase.name)) {
-            throw new Error(
-              `In ${RushConstants.commandLineFilename}, the phase "${phase.name}" is specified ` +
-                'more than once.'
-            );
-          }
+    const commandsForPhase: Map<string, Set<IPhasedCommand>> = new Map();
+    const phaseCliques: Map<string, Set<string>> = new Map();
+    if (commandLineJson?.phases) {
+      for (const phase of commandLineJson.phases) {
+        if (this.phases.has(phase.name)) {
+          throw new Error(
+            `In ${RushConstants.commandLineFilename}, the phase "${phase.name}" is specified ` +
+              'more than once.'
+          );
+        }
 
-          if (phase.name.substring(0, EXPECTED_PHASE_NAME_PREFIX.length) !== EXPECTED_PHASE_NAME_PREFIX) {
-            throw new Error(
-              `In ${RushConstants.commandLineFilename}, the phase "${phase.name}"'s name ` +
-                `does not begin with the required prefix "${EXPECTED_PHASE_NAME_PREFIX}".`
-            );
-          }
+        if (phase.name.substring(0, EXPECTED_PHASE_NAME_PREFIX.length) !== EXPECTED_PHASE_NAME_PREFIX) {
+          throw new Error(
+            `In ${RushConstants.commandLineFilename}, the phase "${phase.name}"'s name ` +
+              `does not begin with the required prefix "${EXPECTED_PHASE_NAME_PREFIX}".`
+          );
+        }
+
+        if (phase.name.length <= EXPECTED_PHASE_NAME_PREFIX.length) {
+          throw new Error(
+            `In ${RushConstants.commandLineFilename}, the phase "${phase.name}"'s name ` +
+              `must have characters after "${EXPECTED_PHASE_NAME_PREFIX}"`
+          );
+        }
+
+        const phaseNameWithoutPrefix: string = phase.name.substring(EXPECTED_PHASE_NAME_PREFIX.length);
+        this.phases.set(phase.name, {
+          ...phase,
+          logFilenameIdentifier: this._normalizeNameForLogFilenameIdentifiers(phase.name),
+          getDisplayNameForProject: (rushProject: RushConfigurationProject) =>
+            `${rushProject.packageName} (${phaseNameWithoutPrefix})`
```

Comment:
It seems like there are only two actual implementations of `getDisplayNameForProject()`, and it's only ever called by `_registerProjectPhaseTask()`.  Could we eliminate this abstraction by testing `isSynthetic` directly in `_registerProjectPhaseTask()`?  Abstractions are confusing.